plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/sonarapi/SonarLintLoggerImpl.java

```diff
@@ -17,36 +17,40 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.api.utils.log;
+package org.sonarsource.sonarlint.core.plugin.commons.sonarapi;
 
+import javax.annotation.Nullable;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.LoggerLevel;
 import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
 
-/**
- * This class can't be moved to another package because {@link BaseLogger} is not public. And we have to extend BaseLogger to please {@link DefaultProfiler}.
- */
-public class SonarApiLogger extends BaseLogger {
+public class SonarLintLoggerImpl implements Logger {
```

Comment:
This class is called `SonarLintLoggerImpl` and has a field with type `SonarLintLogger` but I don't see the relation between the 2 (I was expecting e.g. the 1st one implementing the 2nd one). Maybe one of the names could be improved ?

Reply:
Indeed, I will try something